### ISSUE TITLE
Regarding Equihash implementation 

### ISSUE THREAD

**Issue by jeong760** [bonesoul/CoiniumServ#989]
Dear Bonesoul,

Could you guide me which code should modified if Equihash implemented?

Regards,
John Ahn

**Comment by bonesoul** [comment]
I don't have any idea about equihash algorithm really.

**Comment by jeong760** [comment]
Just let me know which sources are related with handling algorithm and payment for this

**Comment by bonesoul** [comment]
https://github.com/bonesoul/CoiniumServ/tree/89d8c71012972981f97cb80ba4aecb95db7f1572/src/CoiniumServ/Algorithms

**Comment by RepoBike** [comment]
@bonesoul @jeong760 

What about adding ETH to it also?

**Comment by bonesoul** [comment]
eth is a completely different blockchain and coiniumserv won't be able to handle it. though i'm working on  a rewrite.

https://github.com/bonesoul/hypepool

**Comment by jeong760** [comment]
@bonesoul, then Is Equihash able to handling on CoiniumServ without rewriting a code?

**Comment by bonesoul** [comment]
i don't have idea really about equihash.

**Comment by dmshel** [comment]
@jeong760 check ur email box from me
i send u link

**Comment by jeong760** [comment]
@dmshel, Yes  but it seems  to be change many things for the support a Equihash. so I was added  it on algorithm on my repo but  not support a Equihash yet.

**Comment by jeong760** [comment]
@bonesoul, Could you check the following link  for Equihash implementation on CoiniumServ? 
The  following link provide by @dmshel via my personal email. and after I've added a following on 
src/CoiniumServ/Algorithms/ of CoiniumServ  but not fully tested.

- Added an Equihash.cs 

- Modify an AlgorithmRegistry.cs
  - public const string Equihash = "equihash";
  -  // Equihash variants
       _applicationContext.Container.Register<IHashAlgorithm, Equihash>(Equihash).AsMultiInstance(); 

**The CoiniumServ to support an Equihash algo.** 
**Link:** https://github.com/slaxman/ZCoiniumServ

**Comment by bonesoul** [comment]
seems okay.

**Comment by jeong760** [comment]
@Bonesoul, I think I need to modifying a some codes to support and handling Equihash hashing and payment processor

**Comment by dmshel** [comment]
I compiled and launched this version, but there are errors, zcash is working fine, bitcoin gold with errors, but I think this is due to the fact that in the SOL / s engine, the other currencies also gave errors
redis work fine at this release
Log:

03/19/2018 12:45:42 +00:00 [Information] [ConfigManager] [global] CoiniumServ Aurora 0.2.4.13555 warming-up..
03/19/2018 12:45:42 +00:00 [Information] [PlatformManager] [global] Running over Mono 5.10.0.160 (tarball Wed Mar  7 15:25:20 UTC 2018), framework: 4.5 (v4.0.30319.4200
0).
03/19/2018 12:45:42 +00:00 [Information] [ConfigManager] [global] Discovered a total of 2 enabled pool configurations: [Bitcoin Gold, Zcash]
03/19/2018 12:45:42 +00:00 [Information] [MySqlProvider] [Bitcoin Gold] Mysql storage initialized: 127.0.0.1:3306, database: btgcoin.
03/19/2018 12:45:42 +00:00 [Information] [RedisProvider] [Bitcoin Gold] Redis storage initialized: 127.0.0.1:6379, v3.0.6.
03/19/2018 12:45:42 +00:00 [Information] [BanManager] [Bitcoin Gold] No expired bans found to be cleared [remaining bans: 0].
03/19/2018 12:45:42 +00:00 [Information] [JobManager] [Bitcoin Gold] Broadcasted new job 0x1 to 0 subscribers as no new blocks found for last 180 seconds
**03/19/2018 12:45:42 +00:00 [Error] [PaymentProcessor] [Bitcoin Gold] Error getting z address for pool central wallet: GMTETwqhPCVLhbWnvF2aS3n43MV1s7V9gb - Method not fo
und**
03/19/2018 12:45:42 +00:00 [Information] [NetworkInfo] [Bitcoin Gold] symbol: BTG algorithm: equihash version: 150002 protocol: 70016 wallet: 139900 network difficulty:
 3061906.38047999 block difficulty: 3061906.38 network hashrate: 37.61 M SOL/s network: mainnet peers: 9 blocks: 519191 errors: This is a pre-release test build - use a
t your own risk - do not use for mining or merchant applications
03/19/2018 12:45:42 +00:00 [Information] [StratumServer] [Bitcoin Gold] Stratum server listening on 89.111.22.106:3332
03/19/2018 12:45:42 +00:00 [Information] [MySqlProvider] [Zcash] Mysql storage initialized: 127.0.0.1:3306, database: zcashcoin.
03/19/2018 12:45:42 +00:00 [Information] [RedisProvider] [Zcash] Redis storage initialized: 127.0.0.1:6379, v3.0.6.
03/19/2018 12:45:42 +00:00 [Information] [BanManager] [Zcash] No expired bans found to be cleared [remaining bans: 0].
03/19/2018 12:45:42 +00:00 [Information] [JobManager] [Zcash] Broadcasted new job 0x1 to 0 subscribers as no new blocks found for last 180 seconds
03/19/2018 12:45:43 +00:00 [Information] [NetworkInfo] [Zcash] symbol: ZEC algorithm: equihash version: 1001550 protocol: 170003 wallet: 60000 network difficulty: 78118
53.06542592 block difficulty: 7811853.07 network hashrate: 468.12 M SOL/s network: mainnet peers: 8 blocks: 291059 errors: none

**Comment by jeong760** [comment]
@dmshel Which repo do you mean?

**Comment by dmshel** [comment]
@jeong760 
log from https://github.com/slaxman/ZCoiniumServ
i try compiled for test. 
old release and have market module
at ur repo and bonesoul i have error Redis storage initialization failed: 127.0.0.1:6379
and I can not understand what the problem is, I've been reading for several days and am trying various options for correcting the err

**Comment by jeong760** [comment]
@dmshel Yes, it quite old model.

**Comment by dmshel** [comment]
@jeong760 Any news? 
There is a great desire to start testing)

**Comment by igorvoltaic** [comment]
news?

**Comment by devClint** [comment]
@jeong760 did you try the latest redis 3.0.503, also any update on implementing Equihash? If so can you share any updates.

**Comment by devClint** [comment]
@dmshel Hi can you share how you got zcash working please.
